File: backend/src/NLC.Application/DTOs/Planning/PlanningDtos.cs
using NLC.Core.Enums;

namespace NLC.Application.DTOs.Planning;

public record PlanningSlotDto(
    Guid                Id,
    Guid                WarehouseId,
    string              WarehouseName,
    DateOnly            SlotDate,
    TimeOnly            SlotTime,
    JobType             JobType,
    string?             ShipmentDescription,
    string?             ContainerNumber,
    string?             AsnNumber,
    string?             CustomerName,
    string?             DriverName,
    PlanningSlotStatus  Status,
    string?             ErpReference
);

public record CreatePlanningSlotRequest(
    Guid    WarehouseId,
    DateOnly SlotDate,
    TimeOnly SlotTime,
    JobType JobType,
    string? ShipmentDescription,
    string? ContainerNumber,
    string? AsnNumber,
    string? CustomerName,
    string? DriverName,
    string? ErpReference
);

public record UpdatePlanningSlotRequest(
    DateOnly           SlotDate,
    TimeOnly           SlotTime,
    string?            ShipmentDescription,
    string?            ContainerNumber,
    string?            AsnNumber,
    string?            CustomerName,
    string?            DriverName,
    PlanningSlotStatus Status,
    string?            ErpReference
);

File: backend/src/main/java/com/nlc/web/dto/PlanningDtos.java
package com.nlc.web.dto;

import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

public final class PlanningDtos {
    private PlanningDtos() {}

    public record PlanningSlotDto(
        UUID id,
        UUID warehouseId,
        String warehouseName,
        LocalDate slotDate,
        LocalTime slotTime,
        String jobType,
        String shipmentDescription,
        String containerNumber,
        String asnNumber,
        String customerName,
        String driverName,
        String status,
        String erpReference
    ) {}

    public record CreatePlanningSlotRequest(
        @NotNull UUID warehouseId,
        @NotNull LocalDate slotDate,
        LocalTime slotTime,
        String jobType,
        String shipmentDescription,
        String containerNumber,
        String asnNumber,
        String customerName,
        String driverName,
        String erpReference
    ) {}

    public record UpdatePlanningSlotRequest(
        @NotNull LocalDate slotDate,
        LocalTime slotTime,
        String shipmentDescription,
        String containerNumber,
        String asnNumber,
        String customerName,
        String driverName,
        String status,
        String erpReference
    ) {}
}
